v0/destinations/facebook_pixel/transform.js

```diff
@@ -93,45 +95,53 @@ const getContentType = (message, defaultValue, categoryToContent) => {
  *
  * Handles order completed and checkout started types of specific events
  */
-const handleOrder = (message, categoryToContent) => {
-  const { products } = message.properties;
-  const value = formatRevenue(message.properties.revenue);
+const handleOrder = (message, categoryToContent, categoryType) => {
+  const { products, revenue } = message.properties;
+  const value = isDefinedAndNotNull(revenue) ? formatRevenue(revenue) : null;
+
+  if((categoryType === 'order completed') && (!isDefinedAndNotNull(value))){
+    throw new CustomError("Mandatory property revenue is missing from payload. Purchase event could not be sent", 400);
+  }
+
   const contentType = getContentType(message, "product", categoryToContent);
   const contentIds = [];
   const contents = [];
   const { category } = message.properties;
-  if (products && products.length > 0 && Array.isArray(products)) {
-    for (let i = 0; i < products.length; i += 1) {
-      const pId =
-        products[i].product_id || products[i].sku || products[i].id || "";
-      contentIds.push(pId);
-      // required field for content
-      // ref: https://developers.facebook.com/docs/meta-pixel/reference#object-properties
-      const content = {
-        id: pId,
-        quantity: products[i].quantity || message.properties.quantity || 1,
-        item_price: products[i].price || message.properties.price
-      };
-      contents.push(content);
-    }
-    contents.forEach((content, index) => {
-      if (content.id === "") {
-        throw new CustomError(
-          `Product id is required for product ${index}. Event not sent`,
-          400
-        );
+  if(products) {
```

Comment:
What is the logic if products is not there won't we though error?

Reply:
Correct!! we will not, because in Facebook product information is not kept mandatory. But just currency and value will do the job.